lib/tasks/server/express-server.js

```diff
@@ -153,29 +153,30 @@ class ExpressServerTask extends Task {
     this.scheduleServerRestart();
   }
 
-  restartHttpServer() {
+  async restartHttpServer() {
     if (!this.serverRestartPromise) {
-      this.serverRestartPromise = this.stopHttpServer()
-        .then(() => {
+      try {
```

Comment:
This seems wrong to me, the `try`/`catch`/`finally` should be around the async just below now the creation of the `this.serverRestartPromise` chain.

You need something like this:

```js
this.serverRestartPromise = (async () => {
  try {
    await this.stopHttpServer();

    this.emit('restart');
    this.ui.writeLine('');
    this.ui.writeLine(chalk.green('Server restarted.'));
    this.ui.writeLine('');
  } catch(err) {
    this.ui.writeError(err);
  } finally {
    this.serverRestartPromise = null;
  }
})();
```

Reply:
Updated!